charts/prefect-agent/templates/deployment.yaml

```diff
@@ -5,7 +5,7 @@ metadata:
   namespace: {{ include "common.names.namespace" . | quote }}
   labels: {{- include "common.labels.standard" . | nindent 4 }}
     app.kubernetes.io/component: agent
-    prefect-version: {{ .Values.agent.image.prefectTag }}
+    prefect-version: {{ .Chart.AppVersion }}
```

Comment:
Current behavior:

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: prefect-worker
  namespace: "jawnsy-test"
  labels:
    app.kubernetes.io/name: prefect-worker
    helm.sh/chart: prefect-worker-0.0.0
    app.kubernetes.io/instance: worker
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: worker
    prefect-version: 2-python3.11-kubernetes
```

New behavior:

```yaml
apiVersion: v1
kind: Service
metadata:
  name: prefect-server
  namespace: "jawnsy-test"
  labels:
    app.kubernetes.io/name: prefect-server
    helm.sh/chart: prefect-server-0.0.0
    app.kubernetes.io/instance: server
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: server
    prefect-version: 2-latest
```

TODO: we should probably use [`app.kubernetes.io/version`](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/) for this instead